app/models/concept.rb

```diff
@@ -10,4 +10,21 @@ def self.leaf_nodes
                                   .where(concepts[:parent_id].not_eq(nil))
     where.not(concepts[:id].in(distinct_parent_ids))
   end
-end
\ No newline at end of file
+
+  def self.all_with_depth
+    # https://github.com/dockyard/postgres_ext/blob/master/docs/querying.md
+
+    Concept.with.recursive(concepts_tree: <<-SQL
+      SELECT c.id, c.name, c.uid, c.parent_id, 1 as depth
+      FROM concepts c
+      WHERE c.parent_id IS NULL
+
+      UNION
+
+      SELECT c.id, c.name, c.uid, c.parent_id, (concepts_tree.depth + 1)
+      FROM concepts c, concepts_tree
+      WHERE c.parent_id = concepts_tree.id
+    SQL
+    ).select('*').from('concepts_tree')
+  end
```

Comment:
@marcellosachs So did we decide to go with returning `depth` where the lowest depth is the root node? I remember `level` was supposed to start at 0, where the 0 level nodes were all the leaves of the tree. It'd be good to have some clarity on which is the preferred approach going forward.